### ISSUE TITLE
[deploy] Checking Maven Central rules fails for submodules not producing jar archives

### ISSUE THREAD

**Issue by mikemybytes** [jreleaser/jreleaser#1031]
The [current implementation](https://github.com/jreleaser/jreleaser/blob/5babb720a90b69a257bc9fec256f7709e84a112d/sdks/jreleaser-java-sdk-commons/src/main/java/org/jreleaser/sdk/commons/AbstractMavenDeployer.java#L125) of the `AbstractMavenDeployer#checkMavenCentralRules` seems not to be prepared for a multi-module Maven project containing modules that don't produce `jar` archives. For example, verification fails on a submodule with `<packaging>pom</packaging>` with a following message:
```
[nexus2] submodule-with-pom-packaging-0.0.1-sources.jar is missing
(...)
[ERROR] Failed to execute goal org.jreleaser:jreleaser-maven-plugin:1.3.1:full-release (default-cli) on project multi-module-project: Execution default-cli of goal org.jreleaser:jreleaser-maven-plugin:1.3.1:full-release failed: Rules for publishing to Maven Central were not met
(...)
Caused by: org.jreleaser.model.JReleaserException: Rules for publishing to Maven Central were not met
    at org.jreleaser.sdk.commons.AbstractMavenDeployer.collectDeployables (AbstractMavenDeployer.java:116)
    at org.jreleaser.sdk.nexus2.Nexus2MavenDeployer.deploy (Nexus2MavenDeployer.java:54)
    at org.jreleaser.engine.deploy.maven.ProjectMavenDeployer.deploy (ProjectMavenDeployer.java:55)
    at org.jreleaser.engine.deploy.maven.MavenDeployers.deploy (MavenDeployers.java:124)
    at org.jreleaser.engine.deploy.maven.MavenDeployers.deploy (MavenDeployers.java:111)
```

An example project structure causing such issues could be observed [here](https://github.com/mikemybytes/junit5-formatted-source).

Probably, the [Deployable class](https://github.com/jreleaser/jreleaser/blob/5babb720a90b69a257bc9fec256f7709e84a112d/sdks/jreleaser-java-sdk-commons/src/main/java/org/jreleaser/sdk/commons/AbstractMavenDeployer.java#L268)  could be extended with a packaging type info, that could be then used to decide whether or not to expect sources and Javadoc jar archives.

**Comment by aalmiray** [comment]
Oof, absolutely. Staged artifacts with `<packaging>pom</packaging>` (such as BOMs and parent poms) will case rules to fail.

**Comment by aalmiray** [comment]
`Deployable` could add a `packaging` property containing the value of `<packaging>`. This means parsing XML will be a must. Recall that `<packaging>` may not be present, in which case it's default value is `jar`.

Simple XML parsing should be performed otherwise we'd have to resolve parent poms that may also be staged and that would complicate matters a lot.

**Comment by mikemybytes** [comment]
What about a good-old `javax.xml`? A simple XPath query like this should do the trick:
```java
// warning: just a POC
try {
    Document document = DocumentBuilderFactory.newInstance()
                       .newDocumentBuilder()
                       .parse(new File(path));
    String query = "/project/packaging";
    String packaging = (String) XPathFactory.newInstance()
                   .newXPath()
                   .compile(query)
                   .evaluate(document, XPathConstants.STRING);
     // value or an empty string (default packaging)
} catch (ParserConfigurationException | IOException | SAXException | XPathExpressionException e) {
    throw new RuntimeException(e); // TODO: do something meaningful instead
}
```

I'll be happy to pick this up unless you're very into XML parsing 😉

**Comment by aalmiray** [comment]
Thank you @mikemybytes, we welcome contributions! 😄

**Comment by aalmiray** [comment]
Released in v1.4.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.4.0
